viz/metrics-api/proto/viz.proto

```diff
@@ -0,0 +1,537 @@
+  uint64 write_bytes_total = 3;
+}
+
+message TrafficSplitStats {
+  string apex = 2;
+  string leaf = 3;
+  string weight = 4;
+}
+
+message StatTable {
+  oneof table {
+    PodGroup pod_group = 1;
+  }
+
+  message PodGroup {
+    repeated Row rows = 1;
+
+    message Row {
+      Resource resource = 1;
+      string time_window = 2;
+
+      // pod status on Kubernetes
+      string status = 9;
+      // number of pending or running pods in this resource that have linkerd injected
+      uint64 meshed_pod_count = 3;
+      // number of pending or running pods in this resource
+      uint64 running_pod_count = 4;
+      // number of pods in this resource that have Phase PodFailed
+      uint64 failed_pod_count = 6;
+
+      BasicStats stats = 5;
+      TcpStats tcp_stats = 8;
+      TrafficSplitStats ts_stats = 10;
+
+      // Stores a set of errors for each pod name. If a pod has no errors, it may be omitted.
+      map<string, PodErrors> errors_by_pod = 7;
+    }
+  }
+}
+
+message EdgesRequest {
+  ResourceSelection selector = 1;
+}
+
+message EdgesResponse {
+  oneof response {
+    Ok ok = 1;
+    ResourceError error = 2;
+  }
+
+  message Ok {
+    repeated Edge edges = 1;
+  }
+}
+
+message Edge {
+  Resource src = 1;
+  Resource dst = 2;
+  string client_id = 3;
+  string server_id = 4;
+  string no_identity_msg = 5;
+}
+
+message TopRoutesRequest {
+  ResourceSelection selector = 1;
+  string time_window = 2;
+
+  oneof outbound {
+    Empty none = 3;
+    Resource to_resource = 7;
+  }
+}
+
+message TopRoutesResponse {
+  oneof response {
+    ResourceError error = 2;
+    Ok ok = 3;
+  }
+
+  message Ok {
+    repeated RouteTable routes = 1;
+  }
+}
+
+message RouteTable {
+  repeated Row rows = 1;
+  string resource = 2;
+
+  message Row {
+    string route = 1;
+    string time_window = 2;
+    string authority = 6;
+
+    BasicStats stats = 5;
+  }
+}
+
+
+message GatewaysTable {
+  repeated Row rows = 1;
+
+  message Row {
+    string namespace = 1;
+    string name = 2;
+    string cluster_name = 3;
+    uint64 paired_services = 4;
+    bool alive = 5;
+    uint64 latency_ms_p50 = 6;
+    uint64 latency_ms_p95 = 7;
+    uint64 latency_ms_p99 = 8;
+  }
+}
+
+message GatewaysRequest {
+  string remote_cluster_name = 1;
+  string gateway_namespace = 2;
+  string time_window = 3;
+}
+
+message GatewaysResponse {
+  oneof response {
+    Ok ok = 1;
+    ResourceError error = 2;
+  }
+
+  message Ok {
+    GatewaysTable gateways_table = 1;
+  }
+}
+
+service Api {
+  rpc StatSummary(StatSummaryRequest) returns (StatSummaryResponse) {}
+
+  rpc Edges(EdgesRequest) returns (EdgesResponse) {}
+
+  rpc Gateways(GatewaysRequest) returns (GatewaysResponse) {}
+
+  rpc TopRoutes(TopRoutesRequest) returns (TopRoutesResponse) {}
+
+  rpc ListPods(ListPodsRequest) returns (ListPodsResponse) {}
+
+  rpc ListServices(ListServicesRequest) returns (ListServicesResponse) {}
+
+  // Superseded by `TapByResource`.
```

Comment:
Not for this PR but before 2.10 we should investigate if these can be safely removed.